### DISCUSSION TITLE
DML in BAU

### DISCUSSION THREAD

**Discussion by artydev** [grucloud/bau#35] [Ideas]
Hy Frederic,

Thanks for your amazing job :-)
It would be nice to  add DML extension in BAU  like in VANJS 

here is a little demo.

[Van-DML](https://flems.io/#0=N4IgtglgJlA2CmIBcAWAnAOhQDgDQgDMIEBnZAbVADsBDMRJEDACwBcxYR8BjAeytbwByEAB0q4gDxQIANwAE0ALyiQNAA7rVAPikB6GbN0SqXECXgJurCPzKMATEgCMKEAF9c1OgxCyaVBgAVmQ8-ILCjHp68gCCAK6svAC0sZpCUPIE8DSs8QBO8PLiCKzyAPoARjQWuBUkrDTcANbySvLkALoA3PJ98tHyAMqNLVm8+fINExBUAOYV1Rby6rwkEDZ24uKDFmU08lTwAO7yS-Crs2UEE-Ia6hl1JDSyReqFsrbxJPL8ZzXwEZNZriAjxKjWWxUM7wOazAAUAEkACIASnkwHEfXKDWBGHU32Y8KqANR3UGQxeRV4sEy5yxAz0EAI8mJuJaGAQ81YzHk2nkzgADIL0Tz8rxTkdTgBRfLi-Lw1SqMkMknLdrw1gATwevBZSLRbSU7QA5A18rM5ib0QB+eRQXjceL0AQYObwVjShAu1gAIS1iKgBvRSHkKIZhTy+Whavg4ncOxihWmhXksBqZXOdW4AlD3BosFg8jA8VgNnUCHkNnoJFB4Mhfwy8JzZXaznRmOhin1Lb5AvRTd7yX7EY9BWhxPObXqo2a+N46nhqPR-UGyaSqZpmXZIKoCZMgwAKswID8iJW+FR1g0hKxYFr5N94D8AAalF-yZrwLXHCaZWY3jQmR6vIL6Xg0H43JMhRQPE3CWlWzBFJU4JwEU6wAF7wBg2xUIMsSwBANQISQvD0Gc8TEDY0IkFqYCVDSPxJPIMFwUUPLIahlaYdhJQevIADylRBNOQlBPA1h1OUqHwEQRxQHU7y8EkAksu0YkSawboegACuKSTag8qm4aUdwwLEVBQEM3BIbBCACVQABiED5A007wnsTx1GCEJ1McNAbAAsiQ6JKMYfQefxNp2pFrCHhA9C8Ik8I+dwfkBawwWonUUrDB6y4YEBQYhSZ-E2QE7pWY0gi1iYpm4oIenKbw06dn07plKo-iwKoS4YixY7Rohp4YOU3XyJ4uF9HsxR+AWvWyB2DJ9PhhHEfMoE8qeH7Max3DsUhFGWdxEBYThXZ9PV05bSQ2YFAAatOJCjd1y3dqyCgAITGvITr5PdS0Xf0zKss9W73QWRrtH9ANvf0v3MBV8BVbkz7TkVFlWTZyOlvADnOa5rDNoj8zI0CNVeY+6hQKjyJkSVQN9JYywg-CCjGu0YO0hDsConD-TlaTKM1RgUCWB68AeXzjNTC9kPtLI-PPYRN5HK5GBQdKTREkWYVpmzd3-cub37n0k0mH0qj8FKvW8-1N2cqeETwOrBIkDr6Lm-uiZTNV8BNcxp53Cs6azL8wmaYh3wrPpvCqR5fsB7w6JB0E3xlBp1j4rHhm8XhMRKSpBAJ6jScpz88AAI6Ud1t5Vi1hdx8XwDuOiKFlM8rw-M4ZxaoIijQhxxazKdyNHWh52mbwwlJ9Ojfxw1-ux9LfEd3704j6wPNtPy8JtRU5SN+U5ShovSe4Kq3WhpvPMX124MFtfVAbLfDKVLMMjzCQoZdHfl1O0IF238OidDvq3UqZQkh0zANOdm-JZAYCoLwMWh4dRFDtAoUMuVDzwAAB5E0WhAsymQNQOjAHUDAlCbLECgIUKgoVd4MmobSOhGt0xE0RFQOSGwtSog1hMLWNliY0J3vaMihV0iWQAMInlpIqIG89i7MKgKFH6Z9Y7yDtO-SywjaR1DgfIRa8hQxQLIrolRy4-5iLACvKgplGhzB+O0XKTVcFanhPCWg9AKGUJoPkRxDCMQMlMuQJS6hbryEoRgZRdDOhz1jvHPxjjyCCjidFeQIoobhyCLPO0SSfg-xbj4wq-iSCdGCfxMh04HROh9NEwoqMvTwB9J4nw0s+hQVZCE5o+i4kgTEhgW8FpnzwjCSFQG8N6oekEJMdoZDyDNDid9do0lLKyVmGPDBoj5mLNgcYwx2zxF7FiKwVgFp26Sx6YY9pwN9SKLZqozmidl5nA-gbA5etYozIeXUMhxsgbMyKKze5RiObZLLq8nRUTZDkFUGLcJqhQGsiifCkKoivkuzZrCwgiL4QoouOM35ZF-nw0BVMaZmLCFdlNmmfisziHwjIcUmJQgbms1jOiIqk4AR1HyOiQYNBEgpDSA8SydQQZTiQWUZp6htSjijNCfI8Y6jAA6qGTUNA5g5TaaIhxGBtFBlWWLOSyNtX0FROAuq-FzwzN9B-S0TipiiOegah1T19X2q-hrKimLKiiMqKLMiNoMCnikdbTSyNbGmXiNTWm9N3KBP3qZQWlVybPliHKGgD52jkCiSmsmfsylMJJqmwt04jXrPkgyTp8JTJ+pArlIYHpiZI2FumzNWo2G5EChoFKPr8h2ssg65cEz+imWAOUVFRKwDuGnJUN6pkpTQLnRreseLKGToJRgMAvaSH8igHLXm07imbvCdu3d2yMAP15jcvorMl1kXkMs6xt7bmsgfTA76FQZImpUfzPoZCMCFArE0eAAB1DYRIA1VPaKYsAniTjQOXL0MlgHChgF4K8JcdRoOPpWT+jZUAGQ0prVdEC+a20kAzfkLN6IuZQG3pzQ98YiGuvmAAcW4FIrU3AECcOCtOIUwpV6+1RiQQ8vBOOsY-u5fF4TE0VOuPWapW7ViLhuaZZMpZWxZDXXJ56O7Fx7vtIekl714RabLEab9azf3okjOOEBinIWfwWO0CdqLQxTusSY3g0CLPPm09lXTEISO3FrfxH4IFUWjumoWiTnH0bmUstZWyuN8YuQaCXGqCW0pTCsfDPqnzF7ick9wfh+RBFEhtS7QdrmQpPHi2V8tBH5LZX5mxuYnHuO8fgPxhm8MXWescfiQk8JOs3JpQ5oanXA1gBYyYPBqx8hlGNYKqzwAip1AcRExeOGP44dhLMOoGR9xmAsFYTYV4RDOCcA4AA7B4LwIAvG+GCKEEAl4IisBEAlZbZR-DQgIOKGBqgMB6EB+9pUJhwJlGAIYOozBnA4cSEkKgiP8h1FLHUQiE1YEBAwDt7o4hYcYkqEd9H8hHiicELOhWARicQjsJmQwCb-UU8ZXIJcKomdXkgRJZgSDHXYvurEAAcqoOoqhfSxAAKqS9msiQKAAZRFjOSfM9+n4qAwuGT7z6DYO8RRQwmmVwAk0BXfrhFvKGUse9eM1G-qoFWghVCeCrALoX57Fytn5IRTUI7Lc3GUi7fZJow0CCaKwEgJpiPiHKdCOsEIrvDG1AgKR2vWTAENwgbM1uBDeV4CH-IrcgldmTZn9olRDBLkZ5M-iSEgKh7mVz4ADuSBO5AI3sWSqntVg2AgHn8Ma9t-TB3pAqgvu3jd3ngQt4h-9BH+3zvweZkz-GMXhffR8z5APcvs3DRCowEVJ97Xyo6-9G7y7K9EJCJjA1IEgsLsiZX-yCGqgascH4IX02Bf03oQ75EZ7i4SQ5co1LOi3j6rIJahS5dgjBajp7a7Nja4kApKdDSyojnbiwNjXaMDOAADMLgT23g9AIg0SHeZgU+kQTAgBZefQxw0APISA+BDggo6guCF+qw6wV2oYhQ7Ccg8AF+SEEAcwbAoYKAwo7BF+DEu+LsoYzg7BUwNI0A8gcwhQWojO3sJgLAOQPedB-wLQahSUlkoYEwSMQh8AIhYh8g+B6GF+hERwyQwhohrAoYthzSF+gg+CyQBYIhVASA+0c++QF+fAsAEwoYxwJ4ruwB2hVBcODI6gRUlooYgoGAAArHYQtrhPwsXvoVwRsFCKGHcJUKRLAIkIIW-MpEkGACkXXs4dYe4WAPYRsk4ZYS4W4ZkV2NUIYeKKhKGJUOmC0BfgwVAEwUJgAKQhE0jhHyCREbAVFdheGsA+GERzD+GBEzKaHZEu5FB+4QD6EOHwCtFWGuHyAOB2GWwxFYGXZQj2AgDODYBEHuCgIvY+AiCBT3Sy4QxUB1YeBAA)

Regards

**Comment by FredericHeem** [comment]
Adding another way to describe the UI creates more complexity and fragmentation.
Moreover the DML syntax does not support auto-formatting, do it ? 
However feel free to create a repo that implements DML on top of Bau.

**Reply by artydev** [reply]
I understand perfectly.
Indeed DML does not support auto-formatting, and that is effectively not that great.
Thanks Frederic
